server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -199,6 +211,8 @@ public RestoreService(
         }
         this.clusterSettings = clusterService.getClusterSettings();
         this.shardLimitValidator = shardLimitValidator;
+        this.indicesService = indicesService;
+        this.clusterInfoService = clusterInfoService;
```

Comment:
Can we just inject a Supplier<ClusterInfo> instead?

Reply:
Updated.